src/preferences/upgrade.cpp

```diff
@@ -358,21 +359,34 @@ UserSettingsPointer Upgrade::versionUpgrade(const QString& settingsPath) {
 
     if (configVersion.startsWith("1.11")) {
         qDebug() << "Upgrading from v1.11.x...";
-
-        // upgrade to the multi library folder settings
-        QString currentFolder = config->getValueString(PREF_LEGACY_LIBRARY_DIR);
-        // to migrate the DB just add the current directory to the new
-        // directories table
-        TrackCollection tc(config);
-        DirectoryDAO directoryDAO = tc.getDirectoryDAO();
-
-        // NOTE(rryan): We don't have to ask for sandbox permission to this
-        // directory because the normal startup integrity check in Library will
-        // notice if we don't have permission and ask for access. Also, the
-        // Sandbox isn't setup yet at this point in startup because it relies on
-        // the config settings path and this function is what loads the config
-        // so it's not ready yet.
-        bool successful = directoryDAO.addDirectory(currentFolder);
+        bool successful = false;
+        {
+            MixxxDb mixxxDb(config);
+            const mixxx::DbConnectionPool::ThreadLocalScope dbConnectionScope(
```

Comment:
It would be nice to have here a helper for 
const mixxx::DbConnectionPool::ThreadLocalScope dbConnectionScope = mixxxDb.connectionPool().getScopedConnection();